### PR TITLE
Add recipe for esptool

### PR DESCRIPTION
**Pull request by m-rossi**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/esptool/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 1e4288d9f00e55ba36809cc79c493643c623bfa036d7b019a0ebe396284bc317
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  entry_points:
+    - esptool.py = esptool:_main      # [win]
```
Comment (npavlovikj) [inline recipes/esptool/meta.yaml]: Is there a reason why the entry points are used only for Windows? I believe they should be the same for all platforms?

Comment (m-rossi) [reply recipes/esptool/meta.yaml]: I‘ve replicated what’s in their setup.py: https://github.com/espressif/esptool/blob/master/setup.py#L64

Comment (npavlovikj) [reply recipes/esptool/meta.yaml]: Thank you for the clarification!

**Comment by npavlovikj** [comment]
Thank you for contributing @m-rossi 🎉

**Pull request merged**